tests/headless/test_cdn.py

```diff
@@ -0,0 +1,450 @@
+     '/en-US/docs.json?slug=Web/HTML',
+     '/en-US/docs/Web/HTML$json',
+     '/en-US/docs/Web/HTML$history',
+     '/en-US/docs/Web/HTML$children',
+     '/en-US/docs/Web/HTML$revision/1293895',
+     '/en-US/docs/Web/HTML$compare?locale=en-US&to=1299417&from=1293895',
+     '/en-US/Firefox$json',
+     '/en-US/Firefox$history',
+     '/en-US/Firefox$children',
+     '/en-US/docs/Learn/CSS/Styling_text/Fundamentals#Color',
+     '/en-US/docs/Learn/CSS/Styling_text/Fundamentals$toc',
+     '/fr/docs/feeds/rss/l10n-updates',
+     '/fr/docs/localization-tag/inprogress',
+     '/en-US/docs/all',
+     '/en-US/docs/ckeditor_config.js',
+     '/en-US/docs/feeds/atom/files',
+     '/en-US/docs/feeds/rss/all',
+     '/en-US/docs/feeds/rss/needs-review',
+     '/en-US/docs/feeds/rss/needs-review/technical',
+     '/en-US/docs/feeds/rss/revisions',
+     '/en-US/docs/feeds/rss/tag/CSS',
+     '/en-US/docs/needs-review/editorial',
+     '/en-US/docs/tag/ARIA',
+     '/en-US/docs/tags',
+     '/en-US/docs/top-level',
+     '/en-US/docs/with-errors',
+     '/en-US/docs/without-parent'])
+def test_cached(base_url, is_behind_cdn, slug):
+    """Ensure that these requests that should return 200 are cached."""
+    assert_cached(base_url + slug, 200, is_behind_cdn)
+
+
+@pytest.mark.nondestructive
+@pytest.mark.parametrize(
+    'slug', ['/files/2767/hut.jpg',
+             '/@api/deki/files/3613/=hut.jpg',
+             '/en-US/dashboards/localization'])
+def test_cached_301(base_url, is_behind_cdn, slug):
+    """Ensure that these requests that should return 301 are cached."""
+    assert_cached(base_url + slug, 301, is_behind_cdn)
+
+
+@pytest.mark.nondestructive
+@pytest.mark.parametrize(
+    'slug', ['/favicon.ico',
+             '/en-US/events'])
+def test_cached_302(base_url, is_behind_cdn, slug):
+    """Ensure that these requests that should return 302 are cached."""
+    assert_cached(base_url + slug, 302, is_behind_cdn)
+
+
+@pytest.mark.nondestructive
+def test_revisions_dashboard(base_url, is_behind_cdn, kuma_status):
+    """
+    Ensure that the revisions dashboard is cached, and forwards/caches
+    based-on the "X-Requested-With" header and any query parameters.
+    """
+    url = base_url + '/en-US/dashboards/revisions'
+    params = {'page': 2}
+    headers = {'X-Requested-With': 'XMLHttpRequest'}
+    response1 = assert_cached(url, 200, is_behind_cdn)
+    response2 = assert_cached(url, 200, is_behind_cdn, headers=headers)
+    response3 = assert_cached(url, 200, is_behind_cdn, headers=headers,
+                              params=params)
+    assert response3.content != response2.content
+    assert response3.content != response1.content
+    assert response2.content != response1.content
+
+
+@pytest.mark.nondestructive
+@pytest.mark.parametrize(
+    'slug, params',
+    [pytest.param('/en-US/dashboards/topic_lookup', {'topic': 'mathml'},
+                  marks=pytest.mark.skip),  # Gets 504 due to slow response
```

Comment:
What's your reasoning for adding a test that we are skipping from the start?

Reply:
I could go with skip or comment it out. Skip makes it more likely we'll remember to unskip it later, if we improve performance of ``topic_lookup``. A comment means you don't have to update pytest.

Reply:
I'll go the route of updating pytest, thanks @jwhitlock.